streaming/java/streaming-runtime/src/main/java/org/ray/streaming/runtime/transfer/ChannelID.java

```diff
@@ -0,0 +1,194 @@
+package org.ray.streaming.runtime.transfer;
+
+import com.google.common.base.FinalizablePhantomReference;
+import com.google.common.base.FinalizableReferenceQueue;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Sets;
+import com.google.common.io.BaseEncoding;
+import java.lang.ref.Reference;
+import java.nio.ByteBuffer;
+import java.util.Random;
+import java.util.Set;
+import org.ray.runtime.RayNativeRuntime;
+import org.ray.streaming.runtime.util.JniUtils;
+import sun.nio.ch.DirectBuffer;
+
+/**
+ * ChannelID is used to identify a transfer channel between a upstream worker
+ * and downstream worker.
+ */
+public class ChannelID {
+  public static final int ID_LENGTH = 20;
+  private static final FinalizableReferenceQueue REFERENCE_QUEUE = new FinalizableReferenceQueue();
+  // This ensures that the FinalizablePhantomReference itself is not garbage-collected.
+  private static final Set<Reference<?>> references = Sets.newConcurrentHashSet();
+
+  static {
+    // load core_worker_library_java before load streaming_java
+    try {
+      Class.forName(RayNativeRuntime.class.getName());
+    } catch (ClassNotFoundException e) {
+      throw new RuntimeException(e);
+    }
+    JniUtils.loadLibrary("streaming_java");
```

Comment:
Why load the native lib in this `ChannelID` class? This looks a bit weird. Usually we should do this in a runtime class or something.

Reply:
It has some jni methods, it's not sure which class will be loaded first

Reply:
Currently `DataReader`, `DataWriter`, `ChannelID` all have `jni` methods, but no one is an entrypoint class or runtime class. Seems we can put `load libraries` in JobWorker and Tests. I'll try.